router/batchrouter/handle_observability.go

```diff
@@ -128,6 +132,70 @@ func (brt *Handle) recordAsyncDestinationDeliveryStatus(sourceID, destinationID
 	brt.debugger.RecordEventDeliveryStatus(destinationID, &deliveryStatus)
 }
 
+// emitAsyncEventDeliveryTimeMetrics emits event_delivery_time metrics for successful async destination deliveries
+func (brt *Handle) emitAsyncEventDeliveryTimeMetrics(sourceID, destinationID string, statusList []*jobsdb.JobStatusT) {
+	// Get the async destination struct to access original job parameters
+	asyncDestStruct, exists := brt.asyncDestinationStruct[destinationID]
+	if !exists {
+		brt.logger.Debugf("Async destination struct not found for destinationID: %s", destinationID)
+		return
+	}
+
+	// Get destination information for tags
+	destination := asyncDestStruct.Destination
+	if destination == nil {
+		brt.logger.Debugf("Destination not found in async struct for destinationID: %s", destinationID)
+		return
+	}
+
+	destinationTag := misc.GetTagName(destination.ID, destination.Name)
```

Comment:
We can omit the destination.name, I think, as we are adding the destination.ID. And in the later section, we are sending destType.

Reply:
removed it